rama-cli/src/cmd/stunnel/mod.rs

```diff
@@ -0,0 +1,144 @@
+use std::{io::BufReader, path::PathBuf, time::Duration};
+
+use clap::{Args, Subcommand};
+use rama::{
+    Layer,
+    error::{BoxError, ErrorContext, OpaqueError},
+    graceful::Shutdown,
+    service::service_fn,
+    tcp::{TcpStream, server::TcpListener},
+    telemetry::tracing,
+    tls::rustls::{
+        dep::{
+            pemfile,
+            pki_types::{CertificateDer, PrivateKeyDer},
+        },
+        server::{TlsAcceptorDataBuilder, TlsAcceptorLayer},
+    },
+};
+use tokio::io::copy_bidirectional;
+use tracing_subscriber::{
+    EnvFilter, filter::LevelFilter, fmt, layer::SubscriberExt, util::SubscriberInitExt,
+};
+
+#[derive(Debug, Args)]
+/// rama stunnel service
+pub struct StunnelCommand {
+    #[command(subcommand)]
+    pub commands: StunnelSubcommand,
+}
+
+#[derive(Debug, Subcommand)]
+pub enum StunnelSubcommand {
+    Server(ServerArgs),
+    Client(ClientArgs),
+}
+
+#[derive(Debug, Args)]
+pub struct ServerArgs {
+    #[arg(long)]
+    pub listen: String,
```

Comment:
Do look at other commands in this cli to see what are the conventions for args, e.g. https://github.com/plabayo/rama/blob/c377a87aac46c792dccfb276d115e87a4346a472/rama-cli/src/cmd/serve/mod.rs#L37-L39

* try to use the same arguments for the same exact purpose
* try to avoid strings as much as possible, better to immediately parse it as the right type if possible